src/main/cc/common_cpp/protobuf_util/riegeli_io.h

```diff
@@ -36,26 +37,24 @@ namespace wfa {
 // Return error status if the reading fails.
 // The details about the format of the input file can be found in
 // https://github.com/google/riegeli/blob/master/doc/riegeli_records_file_format.md
+// Any message in the file must be stored as google::protobuf::Any, and can be
+// unpacked to the given Message.
 template <typename Message>
 absl::Status ReadRiegeliFile(absl::string_view path,
                              std::vector<Message>& messages) {
   riegeli::RecordReader reader(riegeli::FdReader(path, O_RDONLY));
   messages.clear();
-  while (true) {
-    Message message;
-    if (!reader.ReadRecord(message)) {
-      if (reader.healthy()) {
-        // EOF
-        break;
-      } else {
-        if (!reader.Close()) {
-          return reader.status();
-        }
-        return absl::InvalidArgumentError(
-            absl::StrCat("Unable to read message from file: ", path));
+  google::protobuf::Any any;
+  while (reader.ReadRecord(any)) {
+    if (!any.UnpackTo(&messages.emplace_back())) {
+      messages.pop_back();
+      if (!reader.Close()) {
```

Comment:
There is no need to close RecordReader if we are returning early because of another error. And the error from unpacking Any occurred earlier, so it seems that it should have precedence.